cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -120,6 +120,23 @@
 (rule (lower (has_type ty (iadd_pairwise x y)))
       (addp x y (vector_size ty)))
 
+;;;; Rules for `iabs` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(rule (lower (has_type ty (iabs x)))
```

Comment:
Just a drive-by-comment from me, but I think that the intention is that while this technically works today we'd one day like to add a "chaos mode" of sorts to ISLE selection where any matching rule can be chosen for a lowering. In that sense `iabs` with an `i64` result would match this rule but it's not valid to match this rule since it only works for vectors. I think that this would probably be best to have a matcher/extractor of some sort for the `ty` here like `multi_lane` (or ideally something like `multi_integer_lane` to avoid matching `f64x2` but I don't think we have that yet)

Reply:
Yep, this is a somewhat persistent subtle issue to keep in mind, sorry @dheaton-arm -- each rule has to be correct on its own. One can use rule priorities to make one rule semantically shadowed by others though; that's probably the easiest approach here. So `(rule -1 (lower (has_type ty ...)))` (lower numbers are lower priority).